scipy/stats/tests/test_distributions.py

```diff
@@ -1948,6 +1948,46 @@ def test_fit_simple_non_random_data(self):
         loc, scale = stats.laplace.fit(data, fscale=6)
         assert_allclose(loc, 4, atol=1e-15, rtol=1e-15)
 
+    def test_sf_cdf_extremes(self):
+        # These calculations should not generate warnings.
+        x = 1000
+        p0 = stats.laplace.cdf(-x)
+        # The exact value is smaller than can be represented with
+        # 64 bit floating point, so the exected result is 0.
+        assert p0 == 0.0
+        # The closest 64 bit floating point representation of the
+        # exact value is 1.0.
+        p1 = stats.laplace.cdf(x)
+        assert p1 == 1.0
+
+        p0 = stats.laplace.sf(x)
+        # The exact value is smaller than can be represented with
+        # 64 bit floating point, so the exected result is 0.
+        assert p0 == 0.0
+        # The closest 64 bit floating point represenation of the
+        # exact value is 1.0.
+        p1 = stats.laplace.sf(-x)
+        assert p1 == 1.0
+
+    def test_sf_cdf(self):
+        # Expected value is computed with mpmath:
+        #
+        # >>> import mpmath
+        # >>> mpmath.mp.dps = 60
+        # >>> (mpmath.exp(-200)/2)   # cdf(-200) and sf(200)
+        # 6.919482633683688e-88
+        #
+        expected = 6.919482633683688e-88
+        pneg = stats.laplace.cdf(-200)
+        assert_allclose(pneg, expected, 1e-13)
+        ppos = stats.laplace.sf(200)
+        assert_allclose(ppos, expected, 1e-13)
+
+    def test_isf(self):
+        p = 1e-25
+        x = stats.laplace.isf(p)
+        assert_allclose(x, -np.log(2*p), rtol=1e-13)
+
```

Comment:
This is all fine, they are just a little inconsistent in style. The first checks against mpmath - but could just as well have calculated with NumPy, like the second. (On my machine, it gives precisely the same result in all digits.) The second doesn't check the `ppf` call corresponding with the `isf` call, whereas the first checks that both the `cdf` and `sf` give the expected result. I'd combine the best parts of both - e.g. maybe use NumPy for concision and check `cdf`/`sf` or `ppf`/`isf` pairs.

Reply:
Good point about the use of mpmath here; it isn't actually needed.  I pushed a change to just use numpy to compute the expected values.  I also removed the test of `cdf(-200)`--that would pass in the master branch too, so it isn't really needed.  What's left is two simple tests that would fail in the master branch but pass here.  A "round trip" test could certainly be added, but I don't think we need it.